onedal/decomposition/pca.py

```diff
@@ -51,72 +98,52 @@ def fit(self, X, y=None, queue=None):
         X = _convert_to_supported(policy, X)
 
         params = self.get_onedal_params(X)
-        hparams = get_hyperparameters("covariance", "compute")
-        if hparams is not None and not hparams.is_default:
-            cov_result = self._get_backend(
-                "covariance",
-                None,
-                "compute",
-                policy,
-                {"fptype": params["fptype"], "method": "dense"},
-                hparams.backend,
-                to_table(X),
-            )
-        else:
-            cov_result = self._get_backend(
-                "covariance",
-                None,
-                "compute",
-                policy,
-                {"fptype": params["fptype"], "method": "dense"},
-                to_table(X),
-            )
-        covariance_matrix = from_table(cov_result.cov_matrix)
-        self.mean_ = from_table(cov_result.means)
         result = self._get_backend(
-            "decomposition",
-            "dim_reduction",
-            "train",
-            policy,
-            params,
-            to_table(covariance_matrix),
+            "decomposition", "dim_reduction", "train", policy, params, to_table(X)
```

Comment:
is this the difference between precomputed and cov methods?

Reply:
yes, we don't have to compute covariance separately in "cov" method.